Commenting

samueljohnston96 opened the pull request:
#97

samueljohnston96 committed:
* Comments
* MainPage.cs:
* HostEvent.cs:
* JoinEvent.cs:
* HomeScreen.cs:
* SearchView.cs:
* LibraryView.cs:
* LayoutConsts.cs:
* TestingScreen.cs:
* PlayingScreen.cs:
* SimpleResultList.cs:
* ResultListWrapper.cs:
* ResultListEnumerator.cs:
* Resource.designer.cs:
* ResultListCell.cs:

* JoinEvent.cs: Removed CommandClass since it is no longer needed.

samueljohnston96 committed:
* Comments
* MainPage.cs:
* HostEvent.cs:
* JoinEvent.cs:
* HomeScreen.cs:
* SearchView.cs:
* LibraryView.cs:
* LayoutConsts.cs:
* TestingScreen.cs:
* PlayingScreen.cs:
* SimpleResultList.cs:
* ResultListWrapper.cs:
* ResultListEnumerator.cs:
* Resource.designer.cs:
* ResultListCell.cs:

* JoinEvent.cs: Removed CommandClass since it is no longer needed.

Conflicts:
	MeNext/Layout/HomeScreen.cs
	MeNext/Layout/HostEvent.cs
	MeNext/Layout/JoinEvent.cs
	MeNext/Layout/LayoutConsts.cs
	MeNext/Layout/LibraryView.cs
	MeNext/Layout/MainPage.cs
	MeNext/Layout/PlayingScreen.cs
	MeNext/Layout/ResultList/ResultListCell.cs
	MeNext/Layout/SearchView.cs
	MeNext/net/JoinEvent.cs
	MusicService/ResultListEnumerator.cs
	MusicService/ResultListWrapper.cs
	MusicService/SimpleResultList.cs

samueljohnston96 committed:
More comments

samueljohnston96 committed:
more comments

samueljohnston96 committed:
more comments

samueljohnston96 committed:
Comments

samueljohnston96 committed:
Comments

daniel-centore commented on MeNext.Spotify.Droid/PlayerListener.cs:
```diff
@@ -111,24 +118,28 @@ void OnAuthenticationComplete(AuthenticationResponse response)
 
         // == Callback methods == //
 
+        //Update on login
```
Use `///` comments for method headers. When using `//` comments, you should have a space before text.

daniel-centore commented on MeNext.Spotify.Droid/PlayerListener.cs:
```diff
@@ -111,24 +118,28 @@ void OnAuthenticationComplete(AuthenticationResponse response)
             this.sms.SomethingChanged();
         }
 
+        //Update on logout
```
""

daniel-centore commented on MeNext.Spotify.Droid/PlayerListener.cs:
```diff
@@ -111,24 +118,28 @@ void OnAuthenticationComplete(AuthenticationResponse response)
             this.sms.SomethingChanged();
         }
 
+        //Update when login fails
```
""

daniel-centore commented on MeNext.Spotify.Droid/PlayerListener.cs:
```diff
@@ -111,24 +118,28 @@ void OnAuthenticationComplete(AuthenticationResponse response)
             this.sms.SomethingChanged();
         }
 
+        //Update for temporary errors
```
""

daniel-centore commented on MeNext.Spotify.Droid/SpotifyMusicServiceDroid.cs:
```diff
@@ -85,6 +102,10 @@ public override bool Playing
         }
 
+        /// <summary>
+        /// Gets or sets the playing position.
```
in seconds

daniel-centore commented on MeNext.Spotify.Droid/SpotifyMusicServiceDroid.cs:
```diff
@@ -111,6 +136,10 @@ public override ISong PlayingSong
         }
 
+        /// <summary>
+        /// Gets or sets the volume.
```
on a scale from 0-1

daniel-centore commented on MeNext.Spotify.Droid/SpotifyMusicServiceDroid.cs:
```diff
@@ -134,6 +163,11 @@ public override void Logout()
         }
 
+        /// <summary>
+        /// Plays the song.  Defaults to beginning
```
1 space between sentences

daniel-centore commented on MeNext.Spotify.iOS/SpotifyAuth.cs:
```diff
@@ -82,6 +85,11 @@ public StreamingDelegate CreateStreamingDelegate()
         }
 
+        /// <summary>
+        /// Opens the URL.
```
Wrong. This is called when the app is opened via a URL (namely `me-next://callback`)

daniel-centore commented on MeNext.Spotify.iOS/SpotifyMusicServiceIos.cs:
```diff
@@ -81,6 +86,10 @@ public override bool Playing
         }
 
+        /// <summary>
+        /// Gets or sets the playing position.
```
in seconds

daniel-centore commented on MeNext.Spotify.iOS/SpotifyMusicServiceIos.cs:
```diff
@@ -120,6 +137,10 @@ public override bool LoggedIn
         }
 
+        /// <summary>
+        /// Gets or sets the volume.
```
from 0-1

daniel-centore commented on MeNext.Spotify.iOS/SpotifyMusicServiceIos.cs:
```diff
@@ -139,6 +160,11 @@ public override double Volume
+        /// Plays the song at given position.
+        /// </summary>
+        /// <param name="song">Song.</param>
+        /// <param name="position">Position.  Defaults to beginning, 0</param>
```
in seconds. 1 space btwn sentences.

daniel-centore commented on MeNext.Spotify.iOS/SpotifyMusicServiceIos.cs:
```diff
@@ -158,12 +184,14 @@ public override void PlaySong(ISong song, double position = 0)
             }
         }
 
+        //Login to spotify
```
///

daniel-centore commented on MeNext.Spotify.iOS/SpotifyMusicServiceIos.cs:
```diff
@@ -158,12 +184,14 @@ public override void PlaySong(ISong song, double position = 0)
             SpotifyAuth.Login();
         }
 
+        //Logout of spotify
```
///

daniel-centore commented on MeNext.Spotify.iOS/SpotifyResults/SpotifySimpleResultList.cs:
```diff
@@ -42,6 +42,10 @@ public PageErrorType Error
         }
 
+        /// <summary>
+        /// Gets the first result index.
```
Gets the index of the first result w/in the entire list

daniel-centore commented on MeNext.Spotify.iOS/SpotifyResults/SpotifySimpleResultList.cs:
```diff
@@ -50,6 +54,10 @@ public int FirstResult
         }
 
+        /// <summary>
+        /// Gets the last result index.
```
""

daniel-centore commented on MeNext.Spotify.iOS/StreamingDelegate.cs:
```diff
@@ -57,6 +57,7 @@ public StreamingDelegate(SpotifyMusicServiceIos service)
             });
         }
 
+        //Logged in! Update UI to reflect that.
```
///

daniel-centore commented on MeNext.Spotify.iOS/StreamingDelegate.cs:
```diff
@@ -66,16 +67,19 @@ public override void AudioStreamingDidLogin(SPTAudioStreamingController audioStr
             this.service.SomethingChanged();
         }
 
+        //Logged out.  Update the UI accordingly.
```
///
space

daniel-centore commented on MeNext.Spotify.iOS/StreamingDelegate.cs:
```diff
@@ -66,16 +67,19 @@ public override void AudioStreamingDidLogin(SPTAudioStreamingController audioStr
             this.service.SomethingChanged();
         }
 
+        //Error handling.
```
///

daniel-centore commented on MeNext.Spotify.iOS/StreamingDelegate.cs:
```diff
@@ -66,16 +67,19 @@ public override void AudioStreamingDidLogin(SPTAudioStreamingController audioStr
             Debug.WriteLine("*** Error: " + error.Description, "stream");
         }
 
+        //Message handling
```
///

daniel-centore commented on MeNext.Spotify.iOS/StreamingPlaybackDelegate.cs:
```diff
@@ -18,8 +18,10 @@ public StreamingPlaybackDelegate(SpotifyMusicServiceIos service)
 
         }
 
+        //Music has started playing.  Update the service accordingly.
```
///
space

daniel-centore commented on MeNext.Spotify.iOS/StreamingPlaybackDelegate.cs:
```diff
@@ -18,8 +18,10 @@ public StreamingPlaybackDelegate(SpotifyMusicServiceIos service)
+        //Music has started playing.  Update the service accordingly.
         public override void AudioStreamingDidStartPlayingTrack(SPTAudioStreamingController audioStreaming, string trackUri)
         {
+            //Get playing song
```
space after //

daniel-centore commented on MeNext.Spotify.iOS/StreamingPlaybackDelegate.cs:
```diff
@@ -54,19 +56,22 @@ public override void AudioStreamingDidStartPlayingTrack(SPTAudioStreamingControl
             }
         }
 
+        //Handle songs ending.
```
///

daniel-centore replied:
Okay I'm not typing any more of these. Fix them all pls.

daniel-centore commented on MeNext.Spotify/SpotifyMetadata/MusicServiceImpls/SpotifyAlbum.cs:
```diff
@@ -4,6 +4,9 @@
 namespace MeNext.Spotify
 {
+    /// <summary>
+    /// Class that represents an album and spotify metadata.
```
Not "and". An Album is metadata. "Class that represents an album" is all you need.

daniel-centore commented on MeNext.Spotify/SpotifyMetadata/MusicServiceImpls/SpotifyArtist.cs:
```diff
@@ -4,6 +4,9 @@
 namespace MeNext.Spotify
 {
+    /// <summary>
+    /// Class that represents an artist and their spotify metadata.
```
"Class that represents an artist"

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -4,6 +4,9 @@
 namespace MeNext.Spotify
 {
+    /// <summary>
+    /// The Spotify music service class links the signed in Spotify user to MeNext.
```
remove "signed in" b/c this handles sign in as well

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -57,92 +63,166 @@ public IResultList<IPlaylist> UserLibraryPlaylists
         }
-
+        /// <summary>
+        /// Gets the album.
```
Gets **an** album from a spotify uid

daniel-centore replied:
Correct for o/s as well

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -57,92 +63,166 @@ public IResultList<IPlaylist> UserLibraryPlaylists
         }
 
+        /// <summary>
+        /// Adds the status listener.
```
adds **a** status listener

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -57,92 +63,166 @@ public IResultList<IPlaylist> UserLibraryPlaylists
         }
 
+        /// <summary>
+        /// Songs the ends.
```
Seriously? This is just the default text. You obviously didn't read it. I'm not wasting effort on more like this. Check your work.

samueljohnston96 committed:
Fixed some comments

samueljohnston96 committed:
Fixing of comments

daniel-centore commented:
@samueljohnston96 Your new changes look great. Let me know when you've applied the same pattern thru the whole PR.

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -65,76 +84,148 @@ public IResultList<ISong> UserLibrarySongs
         }
 
+        /// <summary>
+        /// Searchs for an album.
```
Searches for albums

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -65,76 +84,148 @@ public IResultList<ISong> UserLibrarySongs
         }
 
+        /// <summary>
+        /// Searchs for an artist.
```
""

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -65,76 +84,148 @@ public IResultList<ISong> UserLibrarySongs
         }
 
+        /// <summary>
+        /// Searchs for a playlists.
```
""

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -65,76 +84,148 @@ public IResultList<ISong> UserLibrarySongs
         }
 
+        /// <summary>
+        /// Searchs a song.
```
""

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -143,6 +234,9 @@ public void SongEnds(string uri)
         }
 
+        /// <summary>
+        /// Something has changed so update Ui accordingly.
```
UI*

daniel-centore commented on MeNext.Spotify/SpotifyMusicService.cs:
```diff
@@ -172,6 +274,10 @@ public bool HasUserLibrary
         }
 
+        /// <summary>
+        /// Gets a value indicating whether this <see cref="T:MeNext.Spotify.SpotifyMusicService"/> can play.
```
can play music*

daniel-centore commented on MeNext/Layout/LibraryView.cs:
```diff
@@ -53,15 +53,21 @@ public LibraryView(MainController mainController)
         }
-
+        /// <summary>
+        /// User's Spotify account's login status has changed.
```
Remove this line. This function is called whenever anything pertaining to the UI changes.

daniel-centore commented on MeNext/Layout/MainPage.cs:
```diff
@@ -63,7 +65,7 @@ public void SomethingChanged()
                         this.Children.Add(page);
                     }
-                } else {
+                } else { // No Event no tabs.
```
Put this comment on the next line. I realise we have no style guide but, if we did, it would say that comments should typically be on their own line

samueljohnston96 replied:
Yeah I wasn't too sure about this one, especially since the else was on the same line as the } everywhere else.  

} else {
// Comment about what else means
Else statement

or

//Comment about what else means
} else {
Else statement?

The second one matches general if statement commenting practice

daniel-centore replied:
```
// Check if in event
if (inEvent) {
    // Sing a song
    SingSong();
}
```

```
// Check if in event and, if so, sing a song
if (inEvent) {
    SingSong();
}
```

```
// Check if in event
if (inEvent) {
    // Sing a song
    SingSong();
} else {
    // Explode
    Explode();
}
```

```
if (inEvent) {
    // If in event, sing a song
    SingSong();
} else {
    // If not in event, explode
    Explode();
}
```

daniel-centore replied:
Obviously don't go crazy with following these patterns, but they should be a rule of thumb

daniel-centore commented on MeNext/Layout/PlayingScreen.cs:
```diff
@@ -4,11 +4,17 @@
 namespace MeNext
 {
+    /// <summary>
+    /// The currently playing song screen.  Shows the currently playing song and other info.
```
Spaces

daniel-centore commented on MeNext/Layout/ResultList/ListsView.cs:
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+﻿//using System;
```
Uncomment this. `using System` should be on every page, even if it's unused.

daniel-centore commented on MeNext/Layout/ResultList/ResultListCell.cs:
```diff
@@ -186,7 +188,12 @@ private void UpdateSuggestionButton()
-
+        /// <summary>
+        /// Handles the suggestion.
+        /// 
```
Remove extra newline

daniel-centore replied:
Also "Handles a press of the suggestion button"

daniel-centore commented on MeNext/Layout/ResultList/ResultListCell.cs:
```diff
@@ -207,7 +214,11 @@ private void HandleSuggestion(SuggestSetting s, ISong song)
         }
-
+        /// <summary>
+        /// Gets the suggest icon.
```
Gets the suggestion button icon.

samueljohnston96 committed:
Final Comment fixes.

samueljohnston96 committed:
Final Comment fixes.

samueljohnston96 commented:
@daniel-centore Applied throughout the whole PR.`

daniel-centore commented on MeNext/Layout/SearchView.cs:
```diff
@@ -6,6 +6,10 @@
 {
+    /// <summary>
+    /// Search view.
+    /// Contains a list of searched songs and an entry for song info to search.
```
and an entry for song info to search. ==> and a search field

daniel-centore commented on MeNext/net/JoinEvent.cs:
```diff
@@ -1,33 +1,34 @@
         public string JSON { get; set; }
         public string PID { get; set; }
         public CreateEventResult EventResult { get; set; }
+        /// <summary>
```
Add a newline above this

daniel-centore commented on MusicService/ResultListWrapper.cs:
```diff
@@ -4,20 +4,32 @@
         }
-
+        /// <summary>
+        /// Fufilling inheritance contract.  Gets enumerator.
```
Spelling

daniel-centore commented on MusicService/SimpleResultList.cs:
```diff
@@ -7,52 +7,78 @@ public class SimpleResultList<T> : IResultList<T>
         }
-
+        /// <summary>
+        /// Fufilling inheritance contract.
```
Spelling

daniel-centore replied:
There are o/s

daniel-centore reviewed (changes requested):
Almost there :)

samueljohnston96 committed:
Merge commit 'e4e58e36e25b9038174a3eaaebdde3fd0412d9b3'

Conflicts:
	MeNext/Layout/PlayingScreen.cs
	MeNext/Layout/QueueView.cs
	MeNext/Layout/ResultList/ResultListCell.cs
	MeNext/net/API.cs

samueljohnston96 committed:
* Comments
* MainPage.cs:
* HostEvent.cs:
* JoinEvent.cs:
* HomeScreen.cs:
* SearchView.cs:
* LibraryView.cs:
* LayoutConsts.cs:
* TestingScreen.cs:
* PlayingScreen.cs:
* SimpleResultList.cs:
* ResultListWrapper.cs:
* ResultListEnumerator.cs:
* Resource.designer.cs:
* ResultListCell.cs:

* JoinEvent.cs: Removed CommandClass since it is no longer needed.

Conflicts:
	MeNext/Layout/PlayingScreen.cs
	MeNext/Layout/ResultList/ResultListCell.cs

samueljohnston96 committed:
More comments

Conflicts:
	MeNext/net/API.cs

samueljohnston96 committed:
Fixed some comments

Conflicts:
	MeNext/net/API.cs

samueljohnston96 committed:
Fixing of comments

Conflicts:
	MeNext/Layout/QueueView.cs

samueljohnston96 committed:
merge issues

samueljohnston96 committed:
Merge conflict resolution

samueljohnston96 committed:
mid conflict commit

samueljohnston96 committed:
Merged?

samueljohnston96 committed:
Merge branch 'Commenting' of https://github.com/me-next/menext-app into Commenting

daniel-centore reviewed (changes requested):
Doesn't compile

samueljohnston96 committed:
should be better?
